src/main/java/org/spongepowered/common/service/scheduler/SchedulerExecutionService.java

```diff
@@ -0,0 +1,296 @@
+            return this.runnable.cancel(mayInterruptIfRunning);
+        }
+
+        @Override
+        public boolean isCancelled() {
+            // It might be externally cancelled, the runnable would never
+            // run in that case
+            return this.runnable.isCancelled() || this.task.getState() == ScheduledTask.ScheduledTaskState.CANCELED;
+        }
+
+        @Override
+        public boolean isDone() {
+            return this.runnable.isDone();
+        }
+
+        @Override
+        public V get() throws InterruptedException, ExecutionException {
+            return this.runnable.get();
+        }
+
+        @Override
+        public V get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException, TimeoutException {
+            return this.runnable.get(timeout, unit);
+        }
+    }
+
+    /**
+     * An extension of the JREs FutureTask that throws exceptions of the
+     * underlying Runnable/Callable, which is the normal behaviour
+     * of {@link Task}s
+     */
+    private static class ThrowingFutureTask<V> extends FutureTask<V> {
+
+        private ThrowingFutureTask(Callable<V> callable) {
+            super(callable);
+        }
+
+        private ThrowingFutureTask(Runnable runnable) {
+            super(runnable, null);
+        }
+
+        @Override
+        protected void done() {
+            super.done();
+
+            if (isCancelled()) return;
+
+            try {
+                get();
+            } catch (InterruptedException ignored) {
+                //Cannot happen since we're in done()
+            } catch (ExecutionException e) {
+                throw Throwables.propagate(e.getCause());
+            }
+        }
+    }
+
+    /**
+     * An extension of the JREs FutureTask that can be repeatedly executed,
+     * required for scheduling on an interval.
+     */
+    private static class RepeatableFutureTask<V> extends ThrowingFutureTask<V> {
+
+        @Nullable private Task owningTask = null;
+
+        protected RepeatableFutureTask(Runnable runnable) {
+            super(runnable);
+        }
+
+        protected void setTask(Task task) {
+            this.owningTask = task;
+
+            // Since it is set after being scheduled, it might have thrown
+            // an exception already
+            if (isDone() && !isCancelled()) {
+                this.owningTask.cancel();
+            }
+        }
+
+        @Override
+        protected void done() {
+            // A repeating task that is done but hasn't been cancelled has
+            // failed exceptionally. Following the contract of
+            // ScheduledExecutorService, this means the task has to be stopped.
+            if (!isCancelled() && this.owningTask != null) {
+                this.owningTask.cancel();
+            }
+
+            // The ThrowingFutureTask will make sure the exception gets thrown,
+            // if there is any.
+            super.done();
```

Comment:
The one thing that I could see being a problem here is that you're making a call to get() when no value has been set.  Would this not make for some really long hangs?